### DISCUSSION TITLE
Map Mouse Buttons in Config

### DISCUSSION THREAD

**Discussion by Joe-Zer0** [helix-editor/helix#10404] [General]
I would like to propose allowing users to set the mappings for the mouse buttons in the config file.  Here is an example:

```
[mouse.normal.editor]
# right click copies text to system clipboard
mouse2 = ":clipboard-yank"
# something like the following would be useful too
# double-click = "miw"

[mouse.normal.gutter]
# disable setting breakpoints with mouse
mouse1 = "no_op"
```

**Comment by willhansen** [comment]
I've been trying to figure out how to bind middle-click to "gd" and the thumb button to "C-o".  Has it just not been implemented as possible yet?

**Comment by pascalkuthe** [comment]
no its not implemented yet, its probably reasonable to add some form of remap capability but I see helix as a primarily keyboard driven editor so I  see mouse-based mapping as pretty low priority

**Comment by Joe-Zer0** [comment]
That's fair.  Thanks for the response!

**Comment by tmpm697** [comment]
I aslo have this interest, I want to select text with left mouse and then copy them to `system` clipboard with right mouse,
or just yank to system clipboard by select text with left mouse.

**Comment by 0x61nas** [comment]
I think that I can work on this, please assign this issue to me if there's no one working on

**Comment by Nilsia** [comment]
0x61nas I do not know if you implemented this feature yet, however if you are not motivated or don't want to do it anymore.

I would be pleased to take this issue.

I think a bit about it, and I believe that the best way to implement it is to pass into selection mode when double-click / triple-click, or even pressing continuously the left click.
For yanking the solution would be to use the command 'yank' instead of the right click (I do not think that is a good idea to overwrite it, as well as the right click open a menu provided by the OS)

**Comment by 0x61nas** [comment]
> however if you are not motivated or don't want to do it anymore

I don't know if I'll have the space to work on this anytime soon

> I would be pleased to take this issue.

it's yours now, go ahead

**Comment by Joe-Zer0** [comment]
In Windows Terminal (not in Helix) the right click behavior is to copy to clipboard if there is a selection and the paste if there is no selection.  There are no menu's inside the terminal.  I think it would be intuitive to mimic this behavior inside of Helix.

**Comment by Nilsia** [comment]
Yes, that right, when the mouse is enabled, there is no menu.
I will do what I can do and give a try to Windows when finished.

**Comment by Joe-Zer0** [comment]
That would be amazing! Right click yank is my most desired feature.

**Comment by Nilsia** [comment]
Hi, I have mostly finished the configuration for mouses.
However, something is missing... Is there a way to take from the config key sequences like 'miw' ?
I am searching for that implementations.
If not present, I can implement it by adding in Mappable a KeySequence structure, where the keys start with a '>'.
for example: 2-left = [">m", ">i", ">w"]

**Comment by the-mikedavis** [comment]
> Is there a way to take from the config key sequences like 'miw' ?

See #1383

**Comment by peterjc** [comment]
RE:

```
# something like the following would be useful too
# double-click = "miw"
```

Mouse double-click for select word was logged as part of #10208.
